src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,107 @@
+import classNames from 'classnames';
+import {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from '../../types/Todo';
+import { DispatchContext } from '../../contexts/StateContext';
+import { ActionTypes } from '../../types/Actions';
+
+type Props = {
+  data: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ data }) => {
+  const { id, completed, title } = data;
+
+  const dispatch = useContext(DispatchContext);
+
+  const [newTitle, setNewTitle] = useState(title);
+  const [isEditingMode, setIsEditingMode] = useState(false);
+
+  const input = useRef<HTMLInputElement>(null);
+
+  const toggleStatus = (targetId: number) => {
+    dispatch({ type: ActionTypes.ToggleStatus, payload: targetId });
+  };
+
+  const deleteTodo = (targetId: number) => {
+    dispatch({ type: ActionTypes.DeleteTodo, payload: targetId });
+  };
+
+  const onTitleChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(event.target.value);
+  };
+
+  const saveEdit = () => {
+    if (title.trim()) {
+      dispatch({
+        type: ActionTypes.UpdateTodo,
+        payload: { id, title: newTitle, completed },
+      });
+    }
+
+    setIsEditingMode(false);
+  };
```

Comment:
Here you should add the logic to delete todo if the title is empty. And also it would be good to do nothing if newTitle === title as well.